django/contrib/admin/utils.py

```diff
@@ -304,9 +304,14 @@ def _get_non_gfk_field(opts, name):
     """
     For historical reasons, the admin app relies on GenericForeignKeys as being
     "not found" by get_field(). This could likely be cleaned up.
+
+    Reverse relations should also be excluded as these aren't attributes of the
+    model (rather something like `foo_set`).
     """
     field = opts.get_field(name)
-    if field.is_relation and field.many_to_one and not field.related_model:
```

Comment:
Hm, you did write that original code. :stuck_out_tongue: 
I guess it could be related to the docs that say, "These attribute (field.many_to_one, etc.) are present on all fields; however, they will only have boolean values (rather than None) if the field is a relation type (Field.is_relation=True)." It seems we use a similar pattern of checking `is_relation` before the field flags elsewhere, so I think it's okay.